packages/react/src/prop-types/requiredIfGivenPropExists.js

```diff
@@ -14,11 +14,13 @@
  */
 export default function requiredIfGivenPropExists(name, propType) {
   return function check(props, propName, componentName, ...rest) {
+    if (props[name] && props[propName]) {
```

Comment:
Thank you for jumping in @tw15egan! From my reading of the code `propType()` doesn't run in the condition of `props[name] && !props[propName]`, where it should. Probably we can simply apply the following patch to the original logic?

```javascript
diff --git a/packages/react/src/prop-types/requiredIfGivenPropExists.js b/packages/react/src/prop-types/requiredIfGivenPropExists.js
index 355481920..7c0ab9ec2 100644
--- a/packages/react/src/prop-types/requiredIfGivenPropExists.js
+++ b/packages/react/src/prop-types/requiredIfGivenPropExists.js
@@ -14,7 +14,7 @@
  */
 export default function requiredIfGivenPropExists(name, propType) {
   return function check(props, propName, componentName, ...rest) {
-    if (__DEV__ && props[name]) {
+    if (__DEV__ && props[name] != null && props[propName] == null) {
       return new Error(
         `You must provide a value for \`${propName}\` in \`${componentName}\` if \`${name}\` exists.`
       );
```

Reply:
Ah, you're right, good catch. Updating now! 💪

Reply:
@asudoh I've updated it to `props[name] == true` instead of `props[name] != null`. 

The reasoning is that I'm not sure if people are toggling this feature like so 

`enableExpando={shouldExpand ? true : false}`

 In this case, `enableExpando` is never null and caused errors each time. Maybe an edge-case, but figured it would make sense to only evaluate it if it is true.

Reply:
Good point @tw15egan! - Given the presumable nature of the custom prop-check, would we want to rename it to `requiredIfGivenPropIsTruthy()`?

Reply:
@asudoh updated 👍